### DISCUSSION TITLE
How can I access authenticated user in computed property ?

### DISCUSSION THREAD

**Discussion by krunaldodiya** [adonisjs/core#3260] [General]
Hello there,

I am new in Adonis, will you please guide how can I get authenticated user info in @computed hook ?

Also is it possible to get auth user info in @afterFind() hooks ?

Basically I need to check if a Post is favorited by currently authenticated user or not

```
export default class Post extend BaseModel {
  @afterFind()
  public static async checkIsFavorited(post: Post){
     post.isFavorited = await auth.user.related('favorites').query().where('post_id', this.id).first() !== null
  }
}
```

**Comment by thetutlage** [comment]
So in simple terms there is no direct way to access the auth object (which is specific to an HTTP request) in models since that's how Node works. There are no shared global state like threaded languages.

However, it is possible using the Async local storage, which is documented here https://docs.adonisjs.com/guides/async-local-storage

Here I am making an assumption that you are new to the Node ecosystem and hence expected it work more like a threaded environment. So I will suggest you to spend some time

- Understanding how event loop works
- How Node.js is different from threaded languages
- Why you have to share data by reference vs having it available globally in any file like PHP.

And then only use Async local storage
